src/cache/cache.rs

```diff
@@ -203,9 +203,31 @@ pub trait Storage: Send + Sync {
 
 /// Get a suitable `Storage` implementation from the environment.
 pub fn storage_from_environment() -> Box<Storage> {
+    let endpoint_env_res = env::var("SCCACHE_ENDPOINT");
+    let mut endpoint = String::from("s3.amazonaws.com");
+    match endpoint_env_res {
+        Ok(ep) => endpoint = ep,
+        Err(e) => warn!("Failed to interpret endpoint: {:?}", e),
+    }
+    let region_env_res = env::var("SCCACHE_REGION");
```

Comment:
I looked again, and I still don't understand why the `GET Bucket/?Location` request in S3 requires you to own the bucket. It would be nice if we could just autodetect that instead. I guess this is just an optimization anyway, since S3 will send your requests to the right place regardless.